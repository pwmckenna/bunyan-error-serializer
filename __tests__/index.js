import serializer from '../src';
import VError from 'verror';

test('new Error()', () => {
    const error = new Error();
    expect(serializer(error)).toMatchSnapshot();
})

test('new VError(new Error())', () => {
    const error = new VError(new Error());
    expect(serializer(error)).toMatchSnapshot();
})

test('new VError({ cause: new Error(), info: { count: 42 } })', () => {
    const error = new VError({ cause: new VError({ cause: new Error('cause') }), info: { count: 42 } });
    expect(serializer(error)).toMatchSnapshot();
});

test('stackless Error', () => {
    const error = new Error();
    delete error.stack;
    expect(serializer(error)).toMatchSnapshot();
});

test('non Errors', () => {
    serializer(false);
    serializer({});
    serializer(() => {});
});
