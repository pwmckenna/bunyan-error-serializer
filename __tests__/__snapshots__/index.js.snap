exports[`test new Error() 1`] = `
Object {
  "cause": null,
  "code": undefined,
  "message": "",
  "name": "Error",
  "signal": undefined,
  "stack": "Object.<anonymous>.test (__tests__/index.js:5:19)
Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:42:32)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
TreeProcessor.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2431:7)
Env.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:780:17)
jasmine2 (node_modules/jest-jasmine2/build/index.js:96:7)
runTest (node_modules/jest-cli/build/runTest.js:50:10)
promise.then (node_modules/jest-cli/build/TestRunner.js:304:14)
<anonymous>
process._tickCallback (internal/process/next_tick.js:188:7)
",
}
`;

exports[`test new VError({ cause: new Error(), info: { count: 42 } }) 1`] = `
Object {
  "cause": Object {
    "cause": Object {
      "cause": null,
      "code": undefined,
      "message": "cause",
      "name": "Error",
      "signal": undefined,
      "stack": undefined,
    },
    "code": undefined,
    "message": ": cause",
    "name": "VError",
    "signal": undefined,
    "stack": undefined,
  },
  "code": undefined,
  "count": 42,
  "message": ": : cause",
  "name": "VError",
  "signal": undefined,
  "stack": "Object.<anonymous>.test (__tests__/index.js:15:19)
Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:42:32)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
TreeProcessor.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2431:7)
Env.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:780:17)
jasmine2 (node_modules/jest-jasmine2/build/index.js:96:7)
runTest (node_modules/jest-cli/build/runTest.js:50:10)
promise.then (node_modules/jest-cli/build/TestRunner.js:304:14)
<anonymous>
process._tickCallback (internal/process/next_tick.js:188:7)
caused by: VError: : cause
    Object.<anonymous>.test (__tests__/index.js:15:75)
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:42:32)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
    QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
    QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    TreeProcessor.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2431:7)
    Env.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:780:17)
    jasmine2 (node_modules/jest-jasmine2/build/index.js:96:7)
    runTest (node_modules/jest-cli/build/runTest.js:50:10)
    promise.then (node_modules/jest-cli/build/TestRunner.js:304:14)
    <anonymous>
    process._tickCallback (internal/process/next_tick.js:188:7)
caused by: Error: cause
    Object.<anonymous>.test (__tests__/index.js:15:131)
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:42:32)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
    QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
    QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    TreeProcessor.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2431:7)
    Env.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:780:17)
    jasmine2 (node_modules/jest-jasmine2/build/index.js:96:7)
    runTest (node_modules/jest-cli/build/runTest.js:50:10)
    promise.then (node_modules/jest-cli/build/TestRunner.js:304:14)
    <anonymous>
    process._tickCallback (internal/process/next_tick.js:188:7)
",
}
`;

exports[`test new VError(new Error()) 1`] = `
Object {
  "cause": Object {
    "cause": null,
    "code": undefined,
    "message": "",
    "name": "Error",
    "signal": undefined,
    "stack": undefined,
  },
  "code": undefined,
  "message": ": ",
  "name": "VError",
  "signal": undefined,
  "stack": "Object.<anonymous>.test (__tests__/index.js:10:19)
Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:42:32)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
TreeProcessor.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2431:7)
Env.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:780:17)
jasmine2 (node_modules/jest-jasmine2/build/index.js:96:7)
runTest (node_modules/jest-cli/build/runTest.js:50:10)
promise.then (node_modules/jest-cli/build/TestRunner.js:304:14)
<anonymous>
process._tickCallback (internal/process/next_tick.js:188:7)
caused by: Error
    Object.<anonymous>.test (__tests__/index.js:10:66)
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:42:32)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    complete (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:384:9)
    QueueRunner.clearStack (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:681:9)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1949:12)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.<anonymous> (node_modules/jest-jasmine2/build/jasmine-async.js:47:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1966:16
    node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1909:9
    Object.fn (node_modules/jest-jasmine2/build/jasmine-async.js:68:11)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    Spec.queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    Spec.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:372:10)
    Object.fn (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2584:37)
    attemptAsync (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1984:24)
    QueueRunner.run (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1939:9)
    QueueRunner.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:1927:10)
    queueRunnerFactory (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:718:35)
    TreeProcessor.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:2431:7)
    Env.execute (node_modules/jest-jasmine2/vendor/jasmine-2.5.2.js:780:17)
    jasmine2 (node_modules/jest-jasmine2/build/index.js:96:7)
    runTest (node_modules/jest-cli/build/runTest.js:50:10)
    promise.then (node_modules/jest-cli/build/TestRunner.js:304:14)
    <anonymous>
    process._tickCallback (internal/process/next_tick.js:188:7)
",
}
`;

exports[`test stackless Error 1`] = `
Object {
  "cause": null,
  "code": undefined,
  "message": "",
  "name": "Error",
  "signal": undefined,
  "stack": undefined,
}
`;
